src/user/user.spec.js

```diff
@@ -80,6 +80,96 @@ describe("user", () => {
     });
   });
 
+  describe(".set() :", () => {
+    withData(
+      {
+        "sets an email": { email: "wayne_modified@vates.fr" },
+        "sets a password": { password: "newPassword" },
+        "sets a permission": { permission: "user" },
+        "sets a preference": {
+          preferences: {
+            filters: {
+              VM: {
+                test: "name_label: test",
+              },
+            },
+          },
+        },
+      },
+      async data => {
+        data.id = await xo.createUser(simpleUser);
+        expect(await xo.call("user.set", data)).toBe(true);
+        expect(omit(await xo.getUser(data.id), "id")).toMatchSnapshot();
+
+        await testConnection({
+          credentials: {
+            email: data.email === undefined ? simpleUser.email : data.email,
+            password:
+              data.password === undefined ? simpleUser.password : data.password,
+          },
+        });
+      }
+    );
+
+    withData(
+      {
+        "fails trying to set an email with a non admin user connection": {
+          email: "wayne_modified@vates.fr",
+        },
+        "fails trying to set a password with a non admin user connection": {
+          password: "newPassword",
+        },
+        "fails trying to set a permission with a non admin user connection": {
+          permission: "user",
+        },
+      },
+      async data => {
+        data.id = await xo.createUser({
+          email: "wayne8@vates.fr",
+          password: "batman8",
+        });
+        await xo.createUser(simpleUser);
+
+        const xoUser = await getConnection({ credentials: simpleUser });
+        await expect(xoUser.call("user.set", data)).rejects.toMatchSnapshot();
+        await xoUser.close();
+      }
+    );
+
+    it("fails trying to set user permission with its own connection", async () => {
+      const userId = await xo.createUser(simpleUser);
+      const xoUser = await getConnection({ credentials: simpleUser });
```

Comment:
@julien-f  I don't know if we will have to connect with a simple user to test some xo methods but i think it will be a good think if we create a helper which close automatically the connection.

Reply:
Yes, I think such a connection would be used quite a lot.

Reply:
@heafalan  Please create a helper which close automatically the opened connection and i think we should also do this test for the super admin because it can cause troubles if he change its permissions  by mistake.